### PR TITLE
Upgrade to comply with scikitlearn >1.3

### PR DESCRIPTION
**Pull request by Richie94**
With scikit learn 1.3 if_delegate_has_method has been removed in favor if if_available. 
See https://github.com/scikit-learn/scikit-learn/issues/20506

Therefore i updated the scikit learn version to  above 1.3, and changed the calls to the ones i think should be used now.

### CODE REVIEW

**Comment by Richie94** [comment]
Should fix https://github.com/eli5-org/eli5/issues/42

**Comment by solegalli** [comment]
Hey @Richie94 

Thanks for the PR with the changes! 

Could you please remove the modified reqs file from this PR? We are updating the library as we speak and we are managing the new dependencies elsewhere. See #45 

@lopuhin this PR will probably not pass the tests, because we have dependencies on older versions of sklearn where we use the boston dataset. I suggest we keep it on hold while we fix the tests.

**Comment by Richie94** [comment]
it has been introduces in sklearn 1.0.0, so it wont run with old versions... im not mainly a python def, so im not sure on how you want to achieve compatibility with old versions.
0.20 is from 2019, its nearly 6 years ago, i think raising the requirements in a new release can be considered...
but ill remove that, so you can handle it like you want to...

**Comment by solegalli** [comment]
Thank you @Richie94 

No problem at all, the next release of eli5 will only be compatible with sklearn 1.5 and above. 

@lopuhin this is good to merge. 

I said to put it on hold, but actually we need these changes to be able to runs the tests to begin with.

**Comment by solegalli** [comment]
@lopuhin @kmike please merge :)

**Comment by lopuhin** [comment]
Thanks @Richie94 and @solegalli , and sorry @solegalli I missed the message, appreciate a reminder

**Comment by solegalli** [comment]
awesome thank you!
